application-analytics-ui/src/main/resources/Analytics/Code/Macros/EntryPages.xml

```diff
@@ -0,0 +1,492 @@
+        <separator>|</separator>
+        <separators>|</separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>Current User|Current Wiki|Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </visibility>
+    </class>
+    <property>
+      <async_cached>0</async_cached>
+    </property>
+    <property>
+      <async_context/>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <code>{{include reference='Analytics.Code.Macros.VelocityMacros'/}}
+
+{{include reference="Licenses.Code.VelocityMacros"/}}
+
+{{velocity}}
+#macro(executeMacro)
+  #displayAnalyticsMacroTitle('Analytics.Code.Macros.EntryPages', 'rendering.macro.entryPages.description', 'rendering.macro.entryPages.name')
+  #getDate()
+  #set ($discard =$xwiki.jsx.use('Analytics.Code.Macros.RowEvolution'))
+  #set ($parameters = $escapetool.url({
+    'resultPage': 'Analytics.Code.Macros.EntryPagesJSON',
+    'date': $date,
+    'period': 'range',
+    'limitEntries': '-1',
+    'translationPrefix': 'analytics.entryPages.header.'
+  }))
+  #set ($liveDataConfig= {
+    'meta': {
+      'propertyDescriptors': [
+        {'id': 'pageTitle', 'displayer': 'html'},
+        {'id': 'entrances', 'displayer': 'html'},
+        {'id': 'visits', 'displayer': 'html'},
+        {'id': 'bounces', 'displayer': 'html'},
+        {'id': 'actions', 'filterable': false, 'sortable': false, 'displayer': 'html'}
+      ],
+      'entryDescriptor': {
+        'idProperty': 'pageTitle'
+      }
+    }
+  })
+  #getAnalyticsMacroId('entryPagesMacro', $macroId)
+
+  {{liveData
+    id="$macroId"
+    properties="pageTitle, entrances, visits, bounces, actions"
+    source='liveTable'
+    sourceParameters="$parameters"
+    sort='entrances:desc'
+  }}$jsontool.serialize($liveDataConfig){{/liveData}}
+
+  #set ($rowEvolutionParams = {
+    'entrances': 'entry_nb_visits',
+    'bounces': 'entry_bounce_count',
+    'bounceRate': 'bounce_rate',
+    'exitRate': 'exit_rate'
+  })
+  #modalRowEvolution($rowEvolutionParams, 'EntryPages', $macroId)
+#end
+#if ($services.licensing.licensor.hasLicensureForEntity(
+  $services.model.createDocumentReference('', ['Analytics', 'Code'], 'ConfigurationClass')))
+  #if ($xwiki.hasAccessLevel('view', 'Analytics.WebHome'))
+    #executeMacro()
+  #else
+    {{error}}{{translation key="analytics.accessDenied" /}}{{/error}}
+  #end
+#else
+  {{error}}#getMissingLicenseMessage('analytics.extension.name'){{/error}}
+#end
+{{/velocity}}</code>
+    </property>
+    <property>
+      <contentDescription/>
+    </property>
+    <property>
+      <contentJavaType/>
+    </property>
+    <property>
+      <contentType>No content</contentType>
+    </property>
+    <property>
+      <defaultCategory>Analytics</defaultCategory>
+    </property>
+    <property>
+      <description/>
+    </property>
+    <property>
+      <id>mostViewedPages</id>
```

Comment:
Is there any reason for not using the same name for the macro and the wiki page that defines the macro? Isn't it confusing to have a page named ``EntryPages`` that defines a ``mostViewedPages`` macro? If the macro displays the most viewed pages then why not use the same name for the page? Or alternatively use ``entryPages`` as macro name, although I find it hard to understand what "entry pages" means.